griptape/drivers/prompt/anthropic_prompt_driver.py

```diff
@@ -31,47 +41,151 @@ class AnthropicPromptDriver(BasePromptDriver):
     )
     top_p: float = field(default=0.999, kw_only=True, metadata={"serializable": True})
     top_k: int = field(default=250, kw_only=True, metadata={"serializable": True})
+    tool_choice: dict = field(default=Factory(lambda: {"type": "auto"}), kw_only=True, metadata={"serializable": False})
+    use_native_tools: bool = field(default=True, kw_only=True, metadata={"serializable": True})
 
     def try_run(self, prompt_stack: PromptStack) -> TextArtifact:
-        response = self.client.messages.create(**self._base_params(prompt_stack))
+        response = self.client.beta.tools.messages.create(**self._base_params(prompt_stack))
 
-        return TextArtifact(value=response.content[0].text)
+        text = response.content[0].text if response.content and response.content[0].type == "text" else ""
+        tool_uses = [content for content in response.content if content.type == "tool_use"]
+
+        if tool_uses:
+            actions = [
+                ActionsArtifact.Action(
+                    tag=tool_use.id,
+                    name=tool_use.name.split("-")[0],
+                    path=tool_use.name.split("-")[1],
+                    input=tool_use.input,
+                )
+                for tool_use in tool_uses
+            ]
+
+            return ActionsArtifact(value=text, actions=actions)
+        else:
+            return TextArtifact(value=text)
 
     def try_stream(self, prompt_stack: PromptStack) -> Iterator[TextArtifact]:
-        response = self.client.messages.create(**self._base_params(prompt_stack), stream=True)
+        response = self.client.beta.tools.messages.create(**self._base_params(prompt_stack), stream=True)
 
         for chunk in response:
-            if chunk.type == "content_block_delta":
-                yield TextArtifact(value=chunk.delta.text)
+            if chunk.type == "content_block_start":
+                content_block = chunk.content_block
+                content_block_type = content_block.type
 
-    def _prompt_stack_to_model_input(self, prompt_stack: PromptStack) -> dict:
+                if content_block_type == "text":
+                    yield TextArtifact(value=chunk.content_block.text)
+                elif content_block_type == "tool_use":
+                    name, path = content_block.name.split("-")
+                    yield ActionChunkArtifact(
+                        value=f"{name}-{path}", index=chunk.index, tag=content_block.id, name=name, path=path
```

Comment:
Its a little confusing to sometimes set `value` to the "function name" (e.g. `name-path`) and sometimes set it to `partial_input`. It feels like we want two different types of chunks.

Or maybe just add some note in the doc string about how to interpret `value` in the context of `ActionChunkArtifact`.